crates/exex/exex/src/manager.rs

```diff
@@ -278,6 +282,83 @@ impl ExExManager {
         self.handle.clone()
     }
 
+    /// Canonicalizes the WAL by inserting inverts of notifications that have non-canonical blocks.
```

Comment:
I think this lacks context about why we're doing this, making all of this very confusing.